### ISSUE TITLE
Using current directory for caching leads to potential data loss

### ISSUE THREAD

**Issue by igoyak** [keras-team/keras-io#682]
When loading data for the Monocular Depth Estimation example, the current directory is used for caching of the dataset when calling `tf.keras.utils.get_file()`: https://github.com/keras-team/keras-io/blob/master/examples/vision/depth_estimation.py#L56

In certain situations, this method removes the cache directory: https://github.com/keras-team/keras/blob/v2.6.0/keras/utils/data_utils.py#L143

By doing an unfortunate ctrl-c, I managed to remove all of my uncommitted data. 

It might be appropriate to set `cache_subdir` to a subdirectory rather than the current directory to prevent this from happening. Because the rest of the code makes implicit assumptions on where the dataset is, there needs to be changes in multiple places. 

Related to this, the check for whether the dataset has been downloaded does not seem to work, since the `/dataset/` directory is never created by this code: https://github.com/keras-team/keras-io/blob/master/examples/vision/depth_estimation.py#L53

**Comment by maitry63** [comment]
Hi @igoyak , Thanks for reporting this and for the detailed investigation.

I tried to reproduce this on a recent environment:

Keras: 3.14.1
TensorFlow: 2.21.0
Python: 3.12

Created a local directory containing unrelated files (important.txt, project folders, git repo contents)
Used:
```
keras.utils.get_file(
    fname="test.tar.gz",
    origin="file://...",
    cache_subdir=os.path.abspath("."),
    extract=True,
)
```
Interrupted execution during extraction.
After interruption:

- No unrelated files were deleted from the working directory
- Existing files and directories (including untracked/uncommitted files and .git/) remained intact
- Only dataset-related artifacts were created (downloaded archive and extraction outputs)

So, this issue is not able to reproduce the reported `data-loss behavior` on current versions of Keras.

Also, this is not a recommended usage of the API, as it causes dataset artifacts to be written directly into the working directory, leading to clutter and reduced reproducibility. Using a dedicated cache subdirectory `(cache_subdir="dataset")` is preferred.
Additionally, the dataset path existence check in the example does not match the extracted directory structure.

Since this issue is not reproducible with the latest versions, I'll close it for now.

If you still encountering this problem, please open a new issue and include:

TensorFlow version
Keras version
Python version
A complete error traceback

Thank you so much.
